File: mostly asked/containsduplicate.java
import java.util.HashMap;

class containsDuplicate {
    public static void main(String[] args) {
        int arr[]={1,2,3,4,1};
        boolean ans = containsduplicate(arr);
        System.out.println(ans);
    }
    public static boolean containsduplicate(int arr[]){
        HashMap<Integer, Integer> hashmap = new HashMap<>();
        for (int i = 0; i < arr.length; i++) {
            if (hashmap.containsKey(arr[i])) {
                return true;
            }
            hashmap.put(arr[i], i);
        }
        return false;
    }
}

File: Leetcode/mostly asked/containsduplicate.py
def containsduplicate(arr):
    n=len(arr)
    hashmap={}
    for i in range(n):
        if arr[i] in hashmap:
            return True
        
        hashmap[arr[i]] = True
    return False
def containsduplicateset(arr):
    seen = set()
    for num in arr:
        if num in seen:
            return True
        seen.add(num)
    return False
if __name__ == "__main__":
    arr=[1,2,3,4,5,2]
    ans=containsduplicate(arr)
    print(ans)
    ans=containsduplicateset(arr)
    print(ans)